packages/core/core/src/plugin/index.ts

```diff
@@ -25,7 +29,13 @@ export interface MockPlugin {
 export interface BuilderPlugin {
   name: string;
   supportedExtensions: string[];
-  build(configFilePath: string): Promise<string[]>;
+  build(
+    configFilePath: string,
+    destDir: string,
+    verbose?: boolean,
+    importHelpers?: boolean,
+  ): Promise<string>;
+  transpile(sourceCode: string): Promise<string>;
 }
```

Comment:
why do we need `build` and `transpile`, what is the difference?
why both return `Promise<string>`?

Reply:
build compiles a file, return the file path to compiled file

transpile compiles source code in-memory and returns transpiled code, which is used in testing

Reply:
updated